tracer/src/Datadog.Trace/PDBs/SourceLink/AzureDevOpsSourceLinkUrlParser.cs

```diff
@@ -0,0 +1,51 @@
+// <copyright file="AzureDevOpsSourceLinkUrlParser.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System;
+using System.Web;
+
+#pragma warning disable CS1570
+
+namespace Datadog.Trace.Pdb.SourceLink;
+
+internal class AzureDevOpsSourceLinkUrlParser : SourceLinkUrlParser
+{
+    /// <summary>
+    /// Extract the git commit sha and repository url from a Azure DevOps SourceLink mapping string.
+    /// For example, for the following SourceLink mapping string:
+    ///     https://test.visualstudio.com/test-org/_apis/git/repositories/my-repo/items?api-version=1.0&versionType=commit&version=dd35903c688a74b62d1c6a9e4f41371c65704db8&path=/*
+    /// It will return:
+    ///     - commit sha: dd35903c688a74b62d1c6a9e4f41371c65704db8
+    ///     - repository URL: https://test.visualstudio.com/test-org/_git/my-repo
+    /// </summary>
+    internal override bool ParseSourceLinkUrl(Uri uri, out string? commitSha, out string? repositoryUrl)
+    {
+        commitSha = null;
+        repositoryUrl = null;
+
+        // Check if the given URI is a valid AzureDevOps SourceLink mapping string.
+        if (!uri.AbsolutePath.Contains("_apis/git/repositories/") ||
+            !uri.Query.Contains("versionType=commit") ||
+            !uri.Query.Contains("version=") ||
+            !uri.Query.Contains("path=/*"))
+        {
+            return false;
+        }
+
+        // Extract the query string of the URI and check if the required parameters exist.
+        var query = HttpUtility.ParseQueryString(uri.Query);
```

Comment:
You can't use `System.Web.HttpUtility` in netcore 🙁 I think you'll need a different implementation here (e.g. there's [QueryHelpers](https://github.com/dotnet/aspnetcore/blob/main/src/Http/WebUtilities/src/QueryHelpers.cs#L121) you may be able to vendor) or, given it _should_ be a standard format, and shouldn't need URL decoding, a simplistic splitting is probably sufficient

Reply:
Fixed in 45b5532cc4b678569a1a7fca0c9483c9353dfb13